### ISSUE TITLE
False detection of ID selector with sass interpolation

### ISSUE THREAD

**Issue by TxHawks** [stylelint/stylelint#952]
> Describe the issue. Is it a bug or a feature request (new rule, new option, etc.)?

A bug where an interpolated sass string at the beginning of a selector throws a false.

> Which rule, if any, is this issue related to?

`selector-no-id'

> What CSS is needed to reproduce this issue?

```scss
$class-name: '.foo';
#{$class-name} {
  color: pink;
}
```

> What stylelint configuration is needed to reproduce this issue?

```json
{
  "rules": {
    "selector-no-id": true
  }
}
```

> Which version of stylelint are you using?

`5.1.0`

> How are you running stylelint: CLI, PostCSS plugin, Node API?

PostCSS plugin through gulp-postcss:
```js
gulp.task('sass:lint', () =>
  gulp.src('scss/**/*.scss')
    .pipe(plumber())
    .pipe(postcss([
      stylelint(),
      reporter({
        clearMessages: true,
        throwError: true,
      }),
    ],
    { syntax: scssSyntax }))
);
```

> Does your issue relate to non-standard syntax (e.g. SCSS, nesting, etc.)?

Yes, it's related to SCSS interpolation.

> What did you expect to happen?

No warnings to be flagged.

> What actually happened (e.g. what warnings or errors you are getting)?

```shell
test.scss
119:7   ⚠  Unexpected id selector (selector-no-id) [stylelint]
```

**Comment by TxHawks** [comment]
Sorry, this is a duplicate of #945
